cmd/tendermint/main.go

```diff
@@ -1,37 +1,49 @@
 package main
 
 import (
-	"os"
-	"path/filepath"
+	"context"
 
-	cmd "github.com/tendermint/tendermint/cmd/tendermint/commands"
+	"github.com/tendermint/tendermint/cmd/tendermint/commands"
 	"github.com/tendermint/tendermint/cmd/tendermint/commands/debug"
-	"github.com/tendermint/tendermint/config"
 	"github.com/tendermint/tendermint/libs/cli"
+	"github.com/tendermint/tendermint/libs/log"
 	"github.com/tendermint/tendermint/node"
 )
 
 func main() {
-	rootCmd := cmd.RootCmd
-	rootCmd.AddCommand(
-		cmd.GenValidatorCmd,
-		cmd.ReIndexEventCmd,
-		cmd.InitFilesCmd,
-		cmd.LightCmd,
-		cmd.ReplayCmd,
-		cmd.ReplayConsoleCmd,
-		cmd.ResetAllCmd,
-		cmd.ResetPrivValidatorCmd,
-		cmd.ShowValidatorCmd,
-		cmd.TestnetFilesCmd,
-		cmd.ShowNodeIDCmd,
-		cmd.GenNodeKeyCmd,
-		cmd.VersionCmd,
-		cmd.InspectCmd,
-		cmd.RollbackStateCmd,
-		cmd.MakeKeyMigrateCommand(),
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	conf, err := commands.ParseConfig()
+	if err != nil {
+		panic(err)
```

Comment:
I realize some of this was pre-existing, but I think these probably ought to be `stdlog.Fatal`, e.g.,

```go
stdlog.Fatalf("Reading configuration file: %v", err)
```

Reply:
I had the same thought, and opted to avoid doing it in the short term, but definitely open to it.

Reply:
Definitely orthogonal to your immediate purpose, and not urgent.